src/axom/core/Array.hpp

```diff
@@ -33,8 +65,8 @@ class Array;
  * \param [in] arr user-supplied Array instance.
  * \return os the updated output stream object.
  */
-template <typename T>
-std::ostream& operator<<(std::ostream& os, const Array<T>& arr);
+template <typename T, IndexType dim>
```

Comment:
Consistency: Please use capital letters for the non-type template, e.g. `DIM` or `NDIMS`.
I don't think we have any specific rules about this, but it would help readers identify the dimension as a template parameter as opposed to a variable.